search-service/src/main/kotlin/com/egm/stellio/search/util/JsonLdUtils.kt

```diff
@@ -0,0 +1,59 @@
+package com.egm.stellio.search.util
+
+import com.egm.stellio.shared.util.AuthContextModel.AUTH_PROP_KIND
+import com.egm.stellio.shared.util.AuthContextModel.AUTH_PROP_USERNAME
+import com.egm.stellio.shared.util.AuthContextModel.AUTH_TERM_KIND
+import com.egm.stellio.shared.util.AuthContextModel.AUTH_TERM_USERNAME
+import com.egm.stellio.shared.util.ExpandedAttributePayload
+import com.egm.stellio.shared.util.JsonLdUtils
+
+object JsonLdUtils {
+
+    /**
+     * Build the expanded subject info.
+     *
+     * For instance:
+     *
+     * "[
+     *   {
+     *     "@type": [
+     *       "https://uri.etsi.org/ngsi-ld/Property"
+     *     ],
+     *     "https://uri.etsi.org/ngsi-ld/hasValue": [
+     *       {
+     *         "https://ontology.eglobalmark.com/authorization#kind": [
+     *              {
+     *                  "@value": "kind"
+     *               }
+     *         ],
+     *         "https://ontology.eglobalmark.com/authorization#username": [
+     *              {
+     *                  "@value": "username"
+     *              }
+     *         ]
+     *       }
+     *     ]
+     *   }
+     * ]
+     */
+    fun buildExpandedSubjectInfo(value: Map<String, String>): ExpandedAttributePayload =
```

Comment:
this will only work is subject is an user, and not if it is a client or a group. build something more generic.

also wondering if not better to keep it into `EntityAccessRights` (very specific to this use case)